src/Interactive/Features/Interactive/Core/InteractiveHost.Service.cs

```diff
@@ -658,55 +759,97 @@ private Script<object> Compile(string text, string path = null)
                     foreach (PortableExecutableReference reference in script.GetCompilation().DirectiveReferences)
                     {
                         // FullPath refers to the original reference path, not the copy:
-                        LoadReference(reference.FilePath, suppressWarnings: false, addReference: false);
+                        LoadReference(reference.FilePath, suppressWarnings: false, addReference: false, options: ref options);
                     }
 
                     return (Script<object>)script;
                 }
             }
 
+            private async Task<TaskResult> ExecuteFileAsync(
+                RemoteAsyncOperation<RemoteExecutionResult> operation,
+                Task<TaskResult> lastTask,
+                Func<ScriptOptions, string> getFullPath)
+            {
+                var result = await ReportUnhandledExceptionIfAny(lastTask).ConfigureAwait(false);
+                var success = false;
+                try
+                {
+                    var fullPath = getFullPath(result.Options);
+                    var executeResult = await ExecuteFileAsync(result, fullPath).ConfigureAwait(false);
```

Comment:
The public ExecuteFileAsync locks, and this method should not be updating any instance state.